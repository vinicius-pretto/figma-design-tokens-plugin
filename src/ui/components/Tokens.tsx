import * as React from "react";
import * as _ from "lodash";
import Token from "../../consts/Token";

const Tokens = ({ title, tokens, onCreate, onUpdate, message }) => {
  const onMouseEnter = (tokenId: string) => {
    document.querySelector(`#btn-${tokenId}`).classList.remove("hidden");
  };

  const onMouseLeave = (tokenId: string) => {
    document.querySelector(`#btn-${tokenId}`).classList.add("hidden");
  };

  return (
    <section className="tokens-section">
      <div className="tokens-section-header">
        <h3 className="section-title">{title}</h3>

        <div
          className="icon-button"
          role="button"
          aria-label="Add token"
          tabIndex={0}
          onClick={onCreate}
        >
          <div className="icon icon--plus"></div>
        </div>
      </div>

      {_.isEmpty(tokens) ? (
        <div className="mt-lg">
          <p className="label">{message}</p>
        </div>
      ) : (
        <div className="row mt-lg flex-wrap">
          {tokens.map((token: Token) => (
            <div
              key={token.id}
              className="row align-items-center"
              onMouseEnter={() => onMouseEnter(token.id)}
              onMouseLeave={() => onMouseLeave(token.id)}
            >
              <button className="color-token">
                <span
                  className="color-token-shape"
                  style={{ backgroundColor: token.value }}
                ></span>
                <span className="mr-sm">{token.name}</span>
              </button>

              <div
                id={`btn-${token.id}`}
                className="icon-button"
                role="button"
                aria-label="Edit"
                tabIndex={0}
                onClick={() => onUpdate(token)}
              >
                <div className="icon icon--adjust"></div>
              </div>
            </div>
          ))}
        </div>
      )}
    </section>
  );
};

export default Tokens;
