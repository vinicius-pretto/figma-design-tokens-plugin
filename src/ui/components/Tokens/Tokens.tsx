import * as React from "react";
import * as _ from "lodash";
import Token from "../../../consts/Token";
import tokenMessenger from "../../messages/tokenMessenger";

interface Props {
  title: string;
  tokens: Array<object>;
  onCreate: any;
  onUpdate: any;
  message: string;
}

const Tokens = ({ title, tokens, onCreate, onUpdate, message }: Props) => {
  return (
    <section className="border-bottom p-sm">
      <div className="tokens-section-header">
        <h3 className="section-title">{title}</h3>

        <div
          className="icon-button"
          role="button"
          aria-label="Add token"
          tabIndex={0}
          onClick={onCreate}
        >
          <div className="icon icon--plus"></div>
        </div>
      </div>

      {_.isEmpty(tokens) ? (
        <p className="label my-0">{message}</p>
      ) : (
        <div className="d-flex flex-row flex-wrap">
          {tokens.map((token: Token) => (
            <div
              key={token.id}
              className="d-flex flex-row align-items-center justify-content-between w-100 color-token-container"
            >
              <button
                className="color-token"
                onClick={() => tokenMessenger.postSetColorTokenMessage(token)}
              >
                <span
                  className="color-token-shape"
                  style={{ backgroundColor: token.value }}
                ></span>
                <span className="mr-sm">{token.name}</span>
              </button>

              <div
                className="icon-button visibility-hidden"
                role="button"
                aria-label="Edit"
                tabIndex={0}
                onClick={() => onUpdate(token)}
              >
                <div className="icon icon--adjust"></div>
              </div>
            </div>
          ))}
        </div>
      )}
    </section>
  );
};

export default Tokens;
