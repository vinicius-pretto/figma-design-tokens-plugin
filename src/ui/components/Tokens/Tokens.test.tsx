import * as React from "react";
import { render } from "@testing-library/react";
import Tokens from "./Tokens";

describe("Tokens", () => {
  it("snapshot", () => {
    const tokens = [
      {
        id: "99dd84d3-a355-4a3b-a061-6bab7a8bf285",
        type: "color",
        name: "color-dark",
        value: "#222",
      },
      {
        id: "757e0617-d68f-498c-af84-99579f0783d4",
        type: "color",
        name: "color-yellow",
        value: "#c00",
      },
    ];
    const { asFragment } = render(
      <Tokens
        title="Colors"
        tokens={tokens}
        onCreate={jest.fn()}
        onUpdate={jest.fn()}
        onDelete={jest.fn()}
        message="No color tokens"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
