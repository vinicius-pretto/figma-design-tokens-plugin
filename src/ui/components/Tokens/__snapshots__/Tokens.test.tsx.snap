// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tokens snapshot 1`] = `
<DocumentFragment>
  <section
    class="border-bottom p-sm"
  >
    <div
      class="tokens-section-header"
    >
      <h3
        class="section-title"
      >
        Colors
      </h3>
      <div
        aria-label="Add token"
        class="icon-button"
        role="button"
        tabindex="0"
      >
        <div
          class="icon icon--plus"
        />
      </div>
    </div>
    <div
      class="d-flex flex-row flex-wrap"
    >
      <div
        class="d-flex flex-row align-items-center justify-content-between w-100 color-token-container"
      >
        <button
          class="color-token"
        >
          <span
            class="color-token-shape"
            style="background-color: rgb(34, 34, 34);"
          />
          <span
            class="mr-sm"
          >
            color-dark
          </span>
        </button>
        <div
          class="d-flex flex-row"
        >
          <div
            aria-label="Edit"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--adjust"
            />
          </div>
          <div
            aria-label="Delete"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--trash"
            />
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-row align-items-center justify-content-between w-100 color-token-container"
      >
        <button
          class="color-token"
        >
          <span
            class="color-token-shape"
            style="background-color: rgb(204, 0, 0);"
          />
          <span
            class="mr-sm"
          >
            color-yellow
          </span>
        </button>
        <div
          class="d-flex flex-row"
        >
          <div
            aria-label="Edit"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--adjust"
            />
          </div>
          <div
            aria-label="Delete"
            class="icon-button visibility-hidden"
            role="button"
            tabindex="0"
          >
            <div
              class="icon icon--trash"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
